translib/db/db.go

```diff
@@ -1115,6 +1117,44 @@ func (d *DB) StartTx(w []WatchKeys, tss []*TableSpec) error {
 		goto StartTxExit
 	}
 
+	e = d.performWatch(w, tss)
+
+StartTxExit:
+
+	if glog.V(3) {
+		glog.Info("StartTx: End: e: ", e)
+	}
+	return e
+}
+
+func (d *DB) AppendWatchTx(w []WatchKeys, tss []*TableSpec) error {
+	if glog.V(3) {
+		glog.Info("AppendWatchTx: Begin: w: ", w, " tss: ", tss)
+	}
+
+	var e error = nil
+
+	// Validate State
+	if d.txState == txStateNone {
+		glog.Error("AppendWatchTx: Incorrect State, txState: ", d.txState)
+		e = errors.New("Transaction has not started")
+		goto AppendWatchTxExit
+	}
+
+	e = d.performWatch(w, tss)
```

Comment:
Why not use "else" here, so you can avoid goto statement ?

Reply:
Renuka, We have used goto throughout the code. Even in the checked in code we have goto. May I know the reason to avoid goto. I think we need to leave this to programmers preference.